src/utils/cacheUtils.ts

```diff
@@ -0,0 +1,36 @@
+export type CacheRecord<T> = T & {
+  lastFetchedTimestamp: number
+  loading: boolean
+}
+
+const getCacheRecord = <T>(item: T) => {
```

Comment:
i'm a little confused about this function. based on the naming i was expecting it to do the actual "getting" from the cache, but it looks like it just formats something as a `CacheRecord` and the lookup is done elsewhere?

Reply:
Yeah it just does conversion and verifies the data is of the correct type. Maybe we can call it `convertToCacheRecord()`?

Reply:
or like `makeCacheRecord` or `toCacheRecord`? `get` just comes with some overloaded assumptions

Reply:
Yeah, what about `asCacheRecord` since its mostly typing